src/libraries/System.Text.Json/gen/JsonSourceGenerator.Emitter.cs

```diff
@@ -238,12 +253,23 @@ private static SourceText GenerateForTypeWithCustomConverter(ContextGenerationSp
                 SourceWriter writer = CreateSourceWriterWithContextHeader(contextSpec);
 
                 string typeFQN = typeMetadata.TypeRef.FullyQualifiedName;
-                string metadataInitSource = $$"""
-                    {{JsonConverterTypeRef}} converter = {{ExpandConverterMethodName}}(typeof({{typeFQN}}), new {{typeMetadata.ConverterType.FullyQualifiedName}}(), {{OptionsLocalVariableName}});
-                    {{JsonTypeInfoReturnValueLocalVariableName}} = {{JsonMetadataServicesTypeRef}}.{{GetCreateValueInfoMethodRef(typeFQN)}} ({{OptionsLocalVariableName}}, converter);
-                    """;
+                string converterFQN = typeMetadata.ConverterType.FullyQualifiedName;
 
-                GenerateTypeInfoProperty(writer, typeMetadata, metadataInitSource);
+                GenerateTypeInfoFactoryHeader(writer, typeMetadata);
+                if (typeMetadata.ConverterIsJsonStringEnumConverter)
+                {
+                    WriteCompilerDirective(writer, "#pragma warning disable IL3050 // The JsonStringEnumConverter constructor is marked RequiresDynamicCode");
```

Comment:
This isn't going to suppress the warning coming from the Native AOT compiler (which only reads IL). It only suppresses it from the Roslyn Analyzer. If you really want to suppress it you need to `[UnconditionallySuppressMessage]` suppress it.

Reply:
So there's no way we can scope the suppression to individual expressions? It needs to be the full method?

Reply:
I presume we'd need to also detect the TFM to determine if UnconditionalSuppressAttribute is available.

Reply:
> So there's no way we can scope the suppression to individual expressions? It needs to be the full method?

In order to suppress it everywhere, yes.

> I presume we'd need to also detect the TFM to determine if UnconditionalSuppressAttribute is available.

Or generate it into the assembly if it isn't available (which I'm not a fan of).

I think we need to find a way to design this so it doesn't introduce warnings.

Reply:
The problem is that we need to support arbitrary `TConverter : JsonStringEnumConverter` types since the class has been designed with extensibility in mind. I was thinking of using the following structure:
```C#
Helpers.Create<MyDerivedClass>();
Helpers.Create<MyBaseClass>(); // Warns regardless

public static class Helpers
{
    [UnconditionalSuppressMessage("AotAnalysis", "IL3050:RequiresDynamicCode", Justification = "Trust me.")]
    public static MyBaseClass Create<T>() where T : MyBaseClass, new() => new T();
}

public class MyDerivedClass : MyBaseClass { }

public class MyBaseClass
{
    [RequiresDynamicCode("oh no")]
    public MyBaseClass()
    {
    }
}
```
But there's nothing I can do seemingly to suppress potential warnings at the helper callsite. Does this mean I would need to emit bespoke factory helpers for every converter type I encounter?

Reply:
The warning is because the `Create<T>` has a generic constrain `new` - which is equivalent to adding `[DAM(PublicParameterlessConstructore)]` - both will force keep the `.ctor` on the T - and in case of `MyBaseClass` it will warn.

You should get another warning in `MyDerivedClass` since it's `.ctor` is also calling the `MyBaseClas..ctor`.